main.go

```diff
@@ -1600,31 +1601,43 @@ func handleGetRepo(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func getRepoFile(giteaBaseURL, adminUsername, adminPassword, owner, repoName, path, ref string) ([]api.ContentsResponse, error) {
-	url := fmt.Sprintf("%s/repos/%s/%s/contents/%s?ref=%s", giteaBaseURL, owner, repoName, path, ref)
+func getRepoFile(giteaBaseURL, adminUsername, adminPassword, owner, repoName, path, ref string) ([]api.ContentsResponse, *errorapi.APIError) {
+	url := fmt.Sprintf("%s/repos/%s/%s/contents/%s", giteaBaseURL, owner, repoName, path)
+	if ref != "" {
+		// If not specified, defaults to the head of the default branch.
+		url = fmt.Sprintf("%s?ref=%s", url, ref)
+	}
+
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		return nil, err
+		return nil, errorapi.WrapError(errorapi.ErrBadRequest, err.Error())
 	}
 
 	req.SetBasicAuth(string(adminUsername), string(adminPassword))
 
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return nil, err
+		return nil, errorapi.WrapError(errorapi.ErrGiteaConnectError, err.Error())
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("HTTP Error: %d", resp.StatusCode)
+		var errapi *errorapi.APIError
+		if resp.StatusCode == 404 {
+			errapi = errorapi.WrapError(errorapi.ErrNotFound, fmt.Sprintf("file does not exist in repo: %s", path))
+		} else {
+			errapi = errorapi.WrapError(errorapi.ErrBadRequest, fmt.Sprintf("downloading content of file in user repo returned unexpected status: %d", resp.StatusCode))
+		}
+		return nil, errapi
 	}
 
 	bodyBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
 		log.Printf("Error reading Gitea response %v", err)
-		return nil, err
+		return nil, errorapi.WrapError(errorapi.ErrRequestReadError, err.Error())
 	}
 
+	// Not documented but this endpoint can return either a single ContentsResponse as its response or an array of them.
```

Comment:
If this endpoint can return multiple files and that's never a valid thing to return I think we should return a detailed error here instead of checking for that in calling functions. We should have a separate function for `getRepoFiles()` in that case that _can_ return multiple files.

Reply:
It is a valid thing to return, as indicated by the return typing of the function. The Gitea API does not document this.

Reply:
I'm saying for _our_ purposes in this function I don't think it makes sense to return multiple files from a `getRepoFile()` function, so we should check for it within this function and return an error if you've requested the file contents of a directory or something. Then we could create a separate function for fetching multiple files.

Reply:
It is being explicitly used to list directories files as well. The naming is as such due to the poor Gitea documentation. I can rename it to be plural if that's what you take issue with.